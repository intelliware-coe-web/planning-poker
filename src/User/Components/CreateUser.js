import loginImg from '../../resources/images/learning_small.jpg';
import '../../index.css';
import React from 'react';
import {connect} from 'react-redux';
import {PostUser} from "../Actions/UserActions";

function CreateUser(props) {

    return (
        <div className="uk-flex uk-flex-middle uk-height-viewport default-background">
            <div className="uk-width-1-1">
                <img src={loginImg} alt="login"/>
            </div>
            <div className="uk-container uk-width-1-1 uk-margin-small">
                <h1>Welcome to planning poker</h1>
                <h3>Create a new user to get started!</h3>
                <form onSubmit={event => {PostNewUser(event)}}>
                    <div className="uk-margin">
                        <div className="uk-inline uk-width-1-2">
                            <span className="uk-form-icon" uk-icon="icon: user"></span>
                            <input name="userInputName" className="uk-input" placeholder="Name" type="text"/>
                        </div>
                        <button className="uk-button uk-button-primary uk-width-1-4">
                            CREATE
                        </button>
                    </div>
                </form>
            </div>
        </div>
    );

    function PostNewUser(event) {
        event.preventDefault();
        props.postUser(event.target.userInputName.value.trim());
    }

}

const MapStateToProps = state => ({
   user: state.user.user,
   error: state.user.error
});

const MapDispatchToProps = dispatch => ({
   postUser: (username) => dispatch(PostUser(username))
});

export default connect(MapStateToProps, MapDispatchToProps)(CreateUser)