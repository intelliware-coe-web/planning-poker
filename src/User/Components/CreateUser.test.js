import CreateUser from './CreateUser';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore(
    {
        user: {
            user: null,
            error: null
        }
    });

describe('CreateUser', () => {
    it('renders correctly', () => {
        const createUser = renderer.create(<Provider store={store}><CreateUser/></Provider>).toJSON();
        expect(createUser).toMatchSnapshot();
    });
});