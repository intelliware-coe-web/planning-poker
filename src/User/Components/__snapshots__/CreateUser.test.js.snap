// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUser renders correctly 1`] = `
<div
  className="uk-flex uk-flex-middle uk-height-viewport default-background"
>
  <div
    className="uk-width-1-1"
  >
    <img
      alt="login"
      src="learning_small.jpg"
    />
  </div>
  <div
    className="uk-container uk-width-1-1 uk-margin-small"
  >
    <h1>
      Welcome to planning poker
    </h1>
    <h3>
      Create a new user to get started!
    </h3>
    <form
      onSubmit={[Function]}
    >
      <div
        className="uk-margin"
      >
        <div
          className="uk-inline uk-width-1-2"
        >
          <span
            className="uk-form-icon"
            uk-icon="icon: user"
          />
          <input
            className="uk-input"
            name="userInputName"
            placeholder="Name"
            type="text"
          />
        </div>
        <button
          className="uk-button uk-button-primary uk-width-1-4"
        >
          CREATE
        </button>
      </div>
    </form>
  </div>
</div>
`;
